xds/internal/balancer/clusterresolver/resource_resolver_eds.go

```diff
@@ -92,13 +93,39 @@ func (er *edsDiscoveryMechanism) OnError(err error) {
 		return
 	}
 
-	er.topLevelResolver.onError(err)
+	if er.logger.V(2) {
+		er.logger.Infof("EDS discovery mechanism for resource %q reported error: %v", er.nameToWatch, err)
+	}
+
+	// If a previously received good configuration exists, continue to use it.
+	// Else report an empty update that would result in the child policy
+	// reporting TRANSIENT_FAILURE (as there would be no priorities or
+	// localities).
+	er.mu.Lock()
+	if er.update != nil {
+		er.mu.Unlock()
+		return
+	}
+	er.update = &xdsresource.EndpointsUpdate{}
```

Comment:
Actually can you move else comment here after rewriting. And can you also mention alongside what I wrote above that this empty address list plumbing behavior only applies to the *first callback being an error* (of either type), and also triggering failover.

Reply:
Updated.